draft-ietf-quic-transport.md

```diff
@@ -4554,9 +4554,11 @@ connection IDs gives clients some assurance that the server received the packet
 and that the Version Negotiation packet was not generated by an off-path
 attacker.
 
-As future versions of QUIC may support Connection IDs larger than the version 1
-limit, Version Negotiation packets could carry Connection IDs that are longer
-than 20 bytes.
+Future versions of QUIC may have different requirements for the lengths of
+connection IDs larger than version 1. In particular, connection IDs might not
+be required to have a minimum lengths, and might be longer than 20 bytes.  A
+server therefore MUST respond with a Version Negotiation packets for all
+lengths of connection IDs allowed by the layout defined here.
```

Comment:
I'm not sure if "MUST send" is the correct phrase here. This would be a contradiction to what we already have in [section 5.2.2](https://quicwg.org/base-drafts/draft-ietf-quic-transport.html#name-server-packet-handling), states, quote: _if a server receives a packet that indicates an unsupported version but is large enough to initiate a new connection for any supported version, the server SHOULD send a Version Negotiation packet as described in Section 6.1_.

I think what we need to say here is that the CID length rules do not apply to other protocol versions, and no more.

Reply:
PS. Therefore I think that this paragraph can be as simple as something like: _Note that version-specific rules for QUIC packets (e.g., limits on connection ID lengths) SHOULD NOT influence the decision of whether or not to send Version Negotiation packets, because they might change between the versions._

Reply:
That works for me also.  (I wasn't sure about the "MUST" for that reason, but I couldn't find the right words.)  A "MUST NOT" would work for your suggestion:

> Version-specific rules for QUIC packets MUST NOT influence a server decision about whether to send a Version Negotiation packet.

Reply:
I don't think we should say a server MUST send a VN. Makes it sound like it's a protocol violation if I don't respond to a flood of packets, each with their own VN packet. For instance, if MsQuic receives a burst of packets from an endpoint, it generally sends a single VN packet in response.